File: VB/DXApplication3/Form1.vb
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Namespace DXApplication3

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            Dim helper As UnboundRowsHelper = New UnboundRowsHelper(propertyGridControl1)
            helper.AddUnboundRow("UnboundInt", GetType(Integer))
            helper.AddUnboundRow("UnboundString", GetType(String))
            helper.AddUnboundRow("UnboundDateTime", GetType(Date))
            propertyGridControl1.RetrieveFields()
        End Sub
    End Class

    Public Class SourceObject
        Inherits Component

        Public Property TestProperty As String
    End Class
End Namespace

File: CS/DXApplication3/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace DXApplication3
{
        public partial class Form1 : Form
        {
            public Form1()
            {
                InitializeComponent();
                UnboundRowsHelper helper = new UnboundRowsHelper(propertyGridControl1);
                helper.AddUnboundRow("UnboundInt", typeof(int));
                helper.AddUnboundRow("UnboundString", typeof(string));
                helper.AddUnboundRow("UnboundDateTime", typeof(DateTime));
                propertyGridControl1.AutoGenerateRows = true;
            }
        }
}
